### ISSUE TITLE
Function add_column doesn't work for primary_key

### ISSUE THREAD

**Issue by BoWuGit** [sqlalchemy/alembic#1232]
**Describe the bug**
I had a table which didn't has primary_key previously, and later I wanted to add a primary_key column, so I used `add_column` function as follows, but the primary_key was not added. Thanks for your attention.

```py
op.add_column("collection1", sa.Column("id", sa.String, primary_key=True))
```

**To Reproduce**
A minimal example is as follows:

```py
"""interaction test

Revision ID: 3143332ab75e
Revises: 63e25a7c5c04
Create Date: 2023-04-27 09:48:23.650748

"""
from alembic import op
import sqlalchemy as sa

revision = '3143332ab75e'
down_revision = '63e25a7c5c04'
branch_labels = None
depends_on = None

def upgrade():
    op.create_table(
        "collection1",
        sa.Column("status", sa.Integer, nullable=False, server_default="1"),
        sa.Column("created_at", sa.Integer, server_default=sa.text("extract(epoch from now())::int")),
        sa.Column("updated_at", sa.Integer, server_default=sa.text("extract(epoch from now())::int")),
        sa.Column("deleted_at", sa.Integer, server_default=sa.text("0")),
    )
    op.add_column("collection1", sa.Column("id", sa.String, primary_key=True))


def downgrade():
    op.drop_table("collection1")
```

**Versions.**
 - OS:  macOS
 - Python: 3.10
 - Alembic: 1.8.0
 - SQLAlchemy: 1.3.6
 - Database:  PostgresSQL 14.7
 - DBAPI: May be [PEP 249 Python Database API Specification v2.0](http://legacy.python.org/dev/peps/pep-0249)? I don't know, sorry.

**Comment by zzzeek** [comment]
I will add a note to the [documentation](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.add_column) now.   constraints such as primary key or check constraints can only be added using constraint directives such as [op.create_primary_key()](https://alembic.sqlalchemy.org/en/latest/ops.html#alembic.operations.Operations.create_primary_key)

**Comment by BoWuGit** [comment]
OK, thanks.

**Comment by zzzeek** [comment]
the database DDL supports "primary key" like here for  postgresql: https://www.postgresql.org/docs/current/sql-altertable.html   so if they want it, let it render and do what it will.

**Comment by CaselIT** [comment]
seems similar to #1780

**Comment by sqla-tester** [comment]
**Mike Bayer** has proposed a fix for this issue in the **main** branch:

**render PRIMARY KEY in add_column** https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/6567

**Comment by zzzeek** [comment]
wow ~~i didnt see that one at all.~~ apparently saw it and commented but it went out the chute

he wants the FK to be inline and not a separate step, err, yuk.   because that means we have to reimplement it all in alembic the way I'm diong with primary key here.

**Comment by seeholza** [comment]
Just FYI, this broke my migration history, since I had explicit `create_primary_key` statements, now clashing with the additional PK. Could this be a breaking change, i.e. major version?

**Comment by zzzeek** [comment]
> Just FYI, this broke my migration history, since I had explicit `create_primary_key` statements, now clashing with the additional PK. Could this be a breaking change, i.e. major version?

how is it that you have `op.add_column(Column(..., primary_key=True))` ?   autogenerate added them with the primary_key=True parameter?

**Comment by zzzeek** [comment]
if autogenerate does this then ill revert it for now.   need to check

**Comment by zzzeek** [comment]
no, it doesnt render primary_key=True

```
    def test_render_add_column_doesnt_render_pk(self):
        op_obj = ops.AddColumnOp(
            "foo", Column("x", Integer, primary_key=True)
        )
        eq_ignore_whitespace(
            autogenerate.render_op_text(self.autogen_context, op_obj),
            "op.add_column('foo', sa.Column('x', sa.Integer(), nullable=False))"
        )

```

that test passes 

so I think your case is an outlier.    it's already released so there's not too much difference between it being 18.2 vs. 19.0 (18.0 just came out for example)

**Comment by RazerM** [comment]
We have the problem in our migrations that removing `primary_key=True` means the type is no longer `SERIAL`:

```python
op.create_table("foo", sa.Column("id", sa.Integer(), nullable=False))
op.create_primary_key(op.f("foo_pkey"), "foo", ["id"])
```

```
ALTER TABLE foo ADD COLUMN id INTEGER NOT NULL;
ALTER TABLE foo ADD CONSTRAINT foo_pkey PRIMARY KEY (id);
```

but using `primary_key=True` bypasses the `naming_convention` since PostgreSQL creates the constraint, which is also not ideal:

```python
op.add_column("foo", sa.Column("id", sa.Integer(), primary_key=True))
```

```
ALTER TABLE foo ADD COLUMN id SERIAL NOT NULL PRIMARY KEY;
```

**Comment by zzzeek** [comment]
yup now *thats* a revert.  just omit this version, 1.18.4 will work the old way and will do this issue some different way

**Comment by sqla-tester** [comment]
**Mike Bayer** has proposed a fix for this issue in the **main** branch:

**revert primary_key=True change and replace with inline_primary_key** https://gerrit.sqlalchemy.org/c/sqlalchemy/alembic/+/6594

**Comment by zzzeek** [comment]
oh wow we had two versions with this change.   OK

**Comment by zzzeek** [comment]
so this horse has left the barn a bit.   if people have migrated to this version of the change then...they'll have to migrate again.

**Comment by CaselIT** [comment]
instead of reverting can't we create a fix that allows both behaviours?

while ugly maybe an `inline_pk=True` flag on add_column instead, so that for cases like this user can opt out of the default behaviour of inline pk?
